### ISSUE TITLE
Show some warnings only in dev environment

### ISSUE THREAD

**Issue by reidbarber** [adobe/react-spectrum#2944]
# 🙋 Feature Request

Brought up in the context of https://github.com/adobe/react-spectrum/pull/2920#discussion_r821914139, we probably need a way to do certain things only in a dev environment (i.e. warnings). This could improve bundle sizes and help with runtime performance in some places.

The scope of this addition could definitely use some discussion, as well as if this is something we want to do.

## 💁 Possible Solution

From Parcel docs: [Development branch removal](https://parceljs.org/features/production/#development-branch-removal)

**Comment by ryo-manba** [comment]
Hey @reidbarber, just to clarify, are we looking to add conditional branching around warning messages like the following to reduce bundle sizes by not including them in production?

```ts
console.warn('onClick is deprecated, please use onPress');
```

Transforming it into something like this?

```ts
if (process.env.NODE_ENV !== 'production') {
  console.warn('onClick is deprecated, please use onPress');
}
```

**Comment by snowystinger** [comment]
Correct, maybe not all warnings, I think we'd need to audit them. We'd probably also want to include any of the code which performs the check as well. So in the example you shared it'd get rid of an if statement in addition to the log

**Comment by ryo-manba** [comment]
@snowystinger 
Absolutely, wrapping the checks along with the warnings inside a condition for non-production environments seems like a good approach:

```ts
if (process.env.NODE_ENV !== 'production') {
  if (flag) {
    console.warn('onClick is deprecated, please use onPress');
  }
}
```

I'll submit a PR to facilitate further discussion on this!

**Comment by snowystinger** [comment]
Related https://github.com/adobe/react-spectrum/issues/6848
